src/timesync/timesyncd-manager.h

```diff
@@ -116,6 +116,12 @@ struct Manager {
         sd_event_source *event_save_time;
         usec_t save_time_interval_usec;
         bool save_on_exit;
+
+        /* Used to coalesce bus PropertiesChanged events */
+        sd_event_source *deferred_link_ntp_server_event_source;
+        sd_event_source *deferred_system_ntp_server_event_source;
+        sd_event_source *deferred_runtime_ntp_server_event_source;
+        sd_event_source *deferred_fallback_ntp_server_event_source;
```

Comment:
please just add a single event source instead of four, and then add a mask field that stores which event ntp server list has been updated. i.e. add a field here `unsigned ntp_server_change_mask;` or so, and then when the runtime servers changed you do:

```c
m->ntp_server_change_mask |= 1U << server->type;
```

from the deferred event source you then send out a change message for all all props for which a bit is set in the mask, and then set the mask to zero.

Reply:
Thanks for reply, I have updated code per your comment, but there is still not signals emit when I change the NTP servers, there should be something I did wrong, please help check again when feel free, many thanks.